### PR TITLE
added an error-handling middleware

### LINKED ISSUE #13: Implement a global error-handling middleware
I have to create a centralised error handing functionality by:

- [ ] Wiring an error handing middleware
- [ ] defining custom error codes and messages for common error scenarios

### PR DESCRIPTION
**Pull request by niyobern**
**What does this PR do?**
This PR adds a global error-handling middleware that will handle all uncaught errors in controllers and api routes

**Description of Task to be completed?**
defined error-handling middleware

**How should this be manually tested?**

- import the middleware function into main file

`import errorHandler from '/middlewares/errorHandler'
`

- add the middleware as the last middleware in the application with: `app.use(errorhandler)`

**Run Commands**
npm run dev
**Any background context you want to provide?**
I am not logging down errors but sending errors to the client

### CODE REVIEW

**Review by wayneleon1 — APPROVED** [review]

**Review by jkarenzi — APPROVED** [review]

**Review by ambroisegithub — COMMENTED** [review]
That's is Look Good!!!

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const app = (0, express_1.default)();
+app.use((0, cors_1.default)());
+app.use((0, morgan_1.default)("dev"));
+app.get("/", (req, res) => {
+    return res.json({ message: "Welcome To The Dynamites backend e-commerce" });
+});
+const PORT = 3000;
+app.listen(PORT, () => {
+    console.log(`The App is running on ${PORT}`);
+});
```
Comment (Habinezajanvier) [inline src/index.js]: Why do we have this one here

**Comment by niyobern** [comment]
I am not the one who wrote this file. I pulled it from develop

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: src/index.js (hunk 1 above)
Comment (niyobern) [reply src/index.js]: It originated from origin/develop when I did git pull

**Pull request merged**